SetupDataPkg/Library/ConfigVariableListLib/ConfigVariableListLib.c

```diff
@@ -14,6 +14,48 @@
 #include <Library/DxeServicesLib.h>
 #include <Library/PcdLib.h>
 #include <Library/ConfigVariableListLib.h>
+#include <Library/SafeIntLib.h>
+
+/**
+  Return the size of the variable list given a NameSize (including null terminator) and DataSize
+
+  @param[in]  NameSize    Size of the CHAR16 name of the config knob including null terminator
+  @param[in]  DataSize    Size of the Data of the config knob
+  @param[out] NeededSize  Size of the variable list based on these inputs. Unchanged if not EFI_SUCCESS returned.
+
+  @retval EFI_INVALID_PARAMETER NeededSize was null
+  @retval EFI_BUFFER_TOO_SMALL  Overflow occurred on addition
+  @retval EFI_SUCCESS           NeededSize contains the variable list size
+**/
+EFI_STATUS
+EFIAPI
+GetVarListSize (
+  IN  UINT32  NameSize,
+  IN  UINT32  DataSize,
+  OUT UINT32  *NeededSize
+  )
+{
+  UINT32         CalculatedSize = sizeof (CONFIG_VAR_LIST_HDR) + sizeof (EFI_GUID) + sizeof (UINT32) + sizeof (UINT32);
+  RETURN_STATUS  Status;
+
+  if (NeededSize == NULL) {
+    return EFI_INVALID_PARAMETER;
+  }
+
+  Status = SafeUint32Add (CalculatedSize, NameSize, &CalculatedSize);
+  if (RETURN_ERROR (Status)) {
+    return EFI_BUFFER_TOO_SMALL;

```

Comment:
just return the status?

Reply:
I know that RETURN_STATUS == EFI_STATUS, but I didn't know if we should rely on that assumption (I'm sure everyone everywhere does...).